src/schema/posts.ts

```diff
@@ -2093,6 +2147,78 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
 
       return { _: true };
     },
+    moderateSourcePosts: async (
+      _,
+      {
+        sourceId,
+        postIds,
+        status,
+        rejectionReason = null,
+        moderatorMessage = null,
+      }: {
+        postIds: string[];
+      } & Pick<
+        SourcePostModeration,
+        'sourceId' | 'status' | 'rejectionReason' | 'moderatorMessage'
+      >,
+      ctx: AuthContext,
+    ) => {
+      const uniquePostIds = Array.from(new Set(postIds));
+      if (
+        !uniquePostIds.length ||
+        uniquePostIds.length > POST_MODERATION_PAGE_SIZE
+      ) {
+        throw new ValidationError('Invalid array of post IDs provided');
+      }
+
+      if (
+        ![
+          SourcePostModerationStatus.Approved,
+          SourcePostModerationStatus.Rejected,
+        ].includes(status)
+      ) {
+        throw new ValidationError('Invalid status provided');
+      }
+
+      const isModerator = await isPrivilegedMember(ctx, sourceId);
+
+      if (!isModerator) {
+        throw new ForbiddenError('Access denied!');
+      }
+
+      const pendingPosts = await ctx.con
+        .getRepository(SourcePostModeration)
+        .find({
+          where: {
+            id: In(uniquePostIds),
+            sourceId,
+            status: SourcePostModerationStatus.Pending,
+          },
+          select: ['id'],
+        });
+
+      if (pendingPosts.length !== uniquePostIds.length) {
+        throw new ValidationError('Some posts are not pending');
+      }
```

Comment:
Don't want to add scope, but should we just ignore those that are not pending. It could break due to race conditions and it feels bad it fails the whole request because one of the post was not pending. For example it could have been another moderator approves one of the posts just before.

Reply:
Updated ✅